numalogic/registry/artifact.py

```diff
@@ -44,17 +45,17 @@ def save(
             dkeys: dynamic key fields as list/tuple of strings
             primary_artifact: primary artifact to be saved
             secondary_artifact: secondary artifact to be saved
+            models_to_retain: number of models to retain in the DB
             metadata: additional metadata surrounding the artifact that needs to be saved
         """
         pass
 
     @abstractmethod
-    def delete(self, skeys: Sequence[str], dkeys: Sequence[str], version: str) -> None:
+    def delete(self, model_key: str, version: str) -> None:
```

Comment:
Why is the signature being changed here? model key consists of skeys and dkeys, correct?